libcxx/include/module.modulemap

```diff
@@ -245,8 +245,15 @@ module std_stdexcept [system] {
   header "stdexcept"
   export *
 }
-module std_stop_token {
+module std_stop_token [system] {
   header "stop_token"
+  private header "__stop_token/atomic_unique_lock.h"
```

Comment:
Shouldn't these still be submodules?

Reply:
Yes generally every header should be in its own submodule. Otherwise including `stop_token` when building with modules will implicitly include all of these headers as well, even if `stop_token` itself doesn't include them.

Reply:
But `<stop_token>` does need these headers.

Reply:
Well yes. The stop_token implementation headers are a bit unique in that stop_token itself directly includes all of them, and none of the other headers do. Something more typical is like __type_traits headers. You can't do like this.
```
module std_type_traits [system] {
  header "type_traits"
  private header "__type_traits/A.h"
  export *
}
```
1. If say `algorithm` includes __type_traits/A.h it will also get type_traits and all of the other headers declared at the same level.
2. `algorithm` can't include __type_traits/A.h because it's private.

So while this is fine for stop_token, it's going to unravel fast when you try to do some of the more significant ones.

Reply:
Oh, OK, I see what you mean. I still think the status quo makes sense for `stop_token` though, then.

Reply:
I would still make submodules for each header even for `stop_token` as a matter of best practice, you should almost never put multiple headers in the same submodule.

Reply:
Submodules only control visibility. If you make a submodule for all of the __stop_token header, it'll still build all of them into a single pcm. It only controls what's visible when something includes __stop_token/atomic_unique_lock.h. i.e. if you have this.

```
module std_stop_token [system] {
  header "stop_token"
  module atomic_unique_lock { private header "__stop_token/atomic_unique_lock.h" }
}
```

Then std_stop_token builds the same as it does the way you have it right now and still cuts down on all the module builds/pcms. The difference is that if something includes __stop_token/atomic_unique_lock.h, then it behaves like non-modular C and you only get __stop_token/atomic_unique_lock.h. But the way you have it, if you build with modules then you _also_ get stop_token and all of the other __stop_token headers, which is pretty unintuitive.

Reply:
I see, thanks for the clarification. This is clearly relevant for how to setup e.g. `type_traits`. However, for `stop_token`, nobody outside of the `stop_token` module can include `__stop_token/atomic_unique_lock.h`, right?

Reply:
As long as you keep it a `private` header, yes that's right.

Reply:
(I believe the original mega module had submodules for all of the implementation headers like I'm suggesting btw)